cfe_internal/recommendations.cf

```diff
@@ -9,6 +9,66 @@ bundle agent MPF_class_recommendations
         if => "cfengine_internal_purge_policies";
 }
 
+@if minimum_version(3.23.0)
+# The preferred location for ignore_interfaces.rx changed from
+# $(sys.inputdir)/ignore_interfaces.rx to $(sys.workdir)/ignore_interfaces.rx in
+# 3.23.0, versions lower than 3.23.0 only look for the file in inputdir and
+# messages relating to the preferred location change are irrelevant.
+
+bundle agent ignore_interfaces_rx_reccomendations
+# @brief Recommend completing migration of ignore_interfaces.rx from inputdir to workdir
+#
+# @description This bundle identifies the presence of ignore_interfaces.rx in
+# inputdir and emits messages about its presence. Additionally, the file is
+# automatically copied if the file is present in inputdir and the class
+# default:mpf_auto_migrate_ignore_interfaces_rx_to_workdir is defined (not
+# defined by default).
+{
+  meta:
+      "tags" slist => { "cfengine_recommends" };
+
+  classes:
+      "ignore_interfaces_in_workdir" -> { "ENT-9402" }
+        if => fileexists( "$(sys.workdir)/ignore_interfaces.rx" );
+
+      "ignore_interfaces_in_inputdir" -> { "ENT-9402" }
+        if => fileexists( "$(sys.inputdir)/ignore_interfaces.rx" );
+
+  files:
+    default:mpf_auto_migrate_ignore_interfaces_rx_to_workdir.ignore_interfaces_in_inputdir::
+      "$(sys.workdir)/ignore_interfaces.rx" -> { "ENT-9402" }
+        copy_from => local_dcp( "$(sys.inputdir)/ignore_interfaces.rx"),
+        comment => concat( "Excluding interfaces should be done outside of the",
+                           " policy input directory so that it's easier to",
+                           " ignore different interfaces on different hosts.");
+
+    default:cfengine_3::
```

Comment:
why cfengine_3 here? Is ignore_interfaces.rx only in version >=3? Not that we do much with version <3.